src/components/UI/Display/MAPI/clusters/ClusterDetail/ClusterDetailDeleteAction.tsx

```diff
@@ -0,0 +1,205 @@
+import { Box, Text } from 'grommet';
+import useDebounce from 'lib/hooks/useDebounce';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import Button from 'UI/Controls/Button';
+import ConfirmationPrompt from 'UI/Controls/ConfirmationPrompt';
+
+import ClusterDetailDeleteActionClusterDetails from './ClusterDetailDeleteActionClusterDetails';
+import ClusterDetailDeleteActionClusterName from './ClusterDetailDeleteActionClusterName';
+
+const NAME_CONFIRMATION_UPDATE_RATE = 250; // In ms.
+
+enum ConfirmationStep {
+  ClusterDetails,
+  ClusterName,
+}
+
+function getConfirmationPromptTitle(step: ConfirmationStep) {
+  switch (step) {
+    case ConfirmationStep.ClusterDetails:
+      return 'Is this the cluster you want to delete?';
+    case ConfirmationStep.ClusterName:
+      return 'Are you sure you want to delete this cluster forever?';
+    default:
+      return '';
+  }
+}
+
+export enum ClusterDetailDeleteActionNameVariant {
+  ID = 'ID',
+  Name = 'name',
+}
+
+interface IClusterDetailDeleteActionProps
+  extends React.ComponentPropsWithoutRef<typeof Box> {
+  name: string;
+  description: string;
+  creationDate: string;
+  workerNodesCount: number;
+  nodePoolsCount: number;
+  userInstalledAppsCount?: number;
+  onDelete: () => void;
+  isLoading?: boolean;
+  variant?: ClusterDetailDeleteActionNameVariant;
```

Comment:
You can configure which term is used to identify a cluster: `Name` or `ID`

Reply:
Very cool! Thanks for making it adapt to the legacy UI.